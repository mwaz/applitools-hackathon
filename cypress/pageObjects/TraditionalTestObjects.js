'use strict';

const elements = {
    logo: () => cy.get('body > div > div > div.logo-w > a > img'),
    logoUrl: () => cy.get('body > div > div > div.logo-w > a'),
    usernameLabel: () => cy.get('body > div > div > form > div:nth-child(1) > label'),
    passwordLabel: () => cy.get('body > div > div > form > div:nth-child(2) > label'),
    usernameField: () => cy.get('#username'),
    passwordField: () => cy.get('#password'),
    usernameIcon: () => cy.get('.form-group .os-icon-user-male-circle'),
    passwordIcon: () => cy.get('.form-group .os-icon-fingerprint'),
    loginButton: () => cy.get('#log-in'),
    rememberMeCheckBox: () => cy.get('body > div > div > form > div.buttons-w > div.form-check-inline > label > input, [type="checkbox"]'),
    twitterIcon: () => cy.get('body > div > div > form > div.buttons-w > div:nth-child(3) > a:nth-child(1) > img'),
    faceBookIcon: () => cy.get('body > div > div > form > div.buttons-w > div:nth-child(3) > a:nth-child(2) > img'),
    linkedInIcon: () => cy.get('body > div > div > form > div.buttons-w > div:nth-child(3) > a:nth-child(3)'),
    alertWarning: () => cy.get('.alert-warning'),
    userDashboard: () => cy.get('.element-balances'),
    amountsHeader: () => cy.get('#amount'),
    shopifyProduct: (positionId) => cy.get(`#transactionsTable > tbody > tr:nth-child(${positionId}) > td.cell-with-media > span`),
    mailChimpServices: (positionId) => cy.get(`#transactionsTable > tbody > tr:nth-child(${positionId}) > td.cell-with-media > span`),
    ebayMarketplace: (positionId) => cy.get(`#transactionsTable > tbody > tr:nth-child(${positionId}) > td.cell-with-media > span`),
    templatesInc: (positionId) => cy.get(`#transactionsTable > tbody > tr:nth-child(${positionId}) > td.cell-with-media > span`),
    stripePaymentProcessing: (positionId) => cy.get(`#transactionsTable > tbody > tr:nth-child(${positionId}) > td.cell-with-media > span`),
    starbucksCoffee: (positionId) => cy.get(`#transactionsTable > tbody > tr:nth-child(${positionId}) > td.cell-with-media > span`),
    expensesButton: () => cy.get('#showExpensesChart'),
    chartCanvas: () => cy.get('#canvas'),
    datasetButton: () => cy.get('#addDataset'),
    firstAdSection: () => cy.get('#flashSale'),
    secondAdSection: () => cy.get('#flashSale2'),
    authHeader: () => cy.get('.auth-header')


};

const expects = {
    usernameAndPasswordLabelsPresent() {
        elements.usernameLabel().should('be.visible');
        elements.passwordLabel().should('be.visible');
    },

    usernameAndPasswordLoginIconsPresent() {
        elements.usernameIcon().should('be.visible');
        elements.passwordIcon().should('be.visible');
    },

    loginButtonPresent() {
        elements.loginButton().should('be.visible')
    },

    rememberMeCheckboxPresent() {
        elements.rememberMeCheckBox().should('be.visible')
    },

    socialMediaIconsPresent() {
        elements.twitterIcon().should('be.visible');
        elements.faceBookIcon().should('be.visible');
        elements.linkedInIcon().should('be.visible');
    },

    logoPresent() {
        elements.logo().should('be.visible');
    },

    dashboardIsPresent() {
        elements.userDashboard().should('be.visible');
    },

    shopifyProductIsPresent({ positionId }) {
        elements.shopifyProduct(positionId).should('have.text', 'Shopify product');
    },
    mailChimpServicesIsPresent({ positionId }) {
        elements.mailChimpServices(positionId).should('have.text', 'MailChimp Services');
    },
    ebayMarketplaceIsPresent({ positionId }) {
        elements.ebayMarketplace(positionId).should('have.text', 'Ebay Marketplace');
    },
    templatesIncIsPresent({ positionId }) {
        elements.templatesInc(positionId).should('have.text', ' Templates Inc');
    },
    stripePaymentProcessingIsPresent({ positionId }) {
        elements.stripePaymentProcessing(positionId).should('have.text', 'Stripe Payment Processing');
    },
    starbucksCoffeeIsPresent({ positionId }) {
        elements.starbucksCoffee(positionId).should('have.text', 'Starbucks coffee');
    },
    canvasChartIsPresent() {
        elements.chartCanvas().should('be.visible');
    },
    matchImageSnapshot(element, snapshotname) {
        cy.wait(1000);
        cy.get(`${element}`).toMatchImageSnapshot({
            "createDiffImage": true,
            "threshold": 0.01,
            "name": `${snapshotname}`,
            "thresholdType": "percent",
        });
    },
    firstAdSectionIsVisible(){
        elements.firstAdSection().should('be.visible')
    },
    secondAdSectionIsVisible(){
        elements.secondAdSection().should('be.visible')
    },
    usernameLabelTextIsCorrect(){
        elements.usernameLabel().should('have.text', 'Username');
    },
    passwordLabelTextIsCorrect(){
        elements.passwordLabel().should('have.text', 'Password');
    },
    loginLabelTextIsCorrect(){
        elements.authHeader().contains('Login Form');
    },
   


};

const actions = {
    clickLoginButton() {
        expects.usernameAndPasswordLabelsPresent();
        expects.loginButtonPresent();
        elements.loginButton().click();
    },
    clickRememberMeCheckBox() {
        expects.rememberMeCheckboxPresent();
        elements.rememberMeCheckBox().click()
            .check().should('be.checked')
    },
    populateUsernameField(username) {
        elements.usernameField().clear().type(username, { force: true });
    },
    clearUsernameField() {
        elements.usernameField().clear();
    },
    populatePasswordField(password) {
        elements.passwordField().clear().type(password, { force: true });
    },
    clickAmountHeader() {
        expects.dashboardIsPresent();
        elements.amountsHeader().click();
    },
    clickCompareExpensesButton() {
        expects.dashboardIsPresent();
        elements.expensesButton().click();
    },
    clickDatasetButton() {
        expects.canvasChartIsPresent();
        elements.datasetButton().click();
    },
    goToAppUrl(){
        cy.visit(`https://demo.applitools.com/hackathonV2.html`);
    },
    goToAppUrlWithAds(){
        cy.visit(`https://demo.applitools.com/hackathonV2.html?showAd=true`);
    }

}

module.exports = {
    expects,
    actions,
    elements
};
