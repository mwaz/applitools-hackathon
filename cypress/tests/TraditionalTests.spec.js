'use strict';
import * as pageObjects from '../pageObjects/TraditionalTestObjects';

describe('ACME app', () => {
  describe('ACME app User login', () => {
    beforeEach(() => {
      pageObjects.actions.goToAppUrl();
    });
    describe('Login Page UI Elements Test:', () => {
    it('can get all form elements on login page', () => {
      pageObjects.expects.logoPresent();
      pageObjects.expects.loginButtonPresent();
      pageObjects.expects.rememberMeCheckboxPresent();
      pageObjects.expects.usernameAndPasswordLabelsPresent();
      pageObjects.expects.usernameAndPasswordLoginIconsPresent();
      pageObjects.expects.socialMediaIconsPresent();
    });

    it('can get correct text for input labels', () => {
      pageObjects.expects.usernameLabelTextIsCorrect();
      pageObjects.expects.passwordLabelTextIsCorrect();
    });

    it('can get correct text headers', () => {
      pageObjects.expects.loginLabelTextIsCorrect(); 
    });

    it('can get correct a snapshot of the logo section', () => {
      cy.get('.logo-w').toMatchSnapshot();
    });
    it('can get correct a snapshot of the form group', () => {
      cy.get('.form-group').toMatchSnapshot();
    });
    it('can get correct a snapshot of the auth header', () => {
      cy.get('.auth-header').toMatchSnapshot();

    });
    it('can get correct a snapshot of the button group', () => {
      cy.get('.buttons-w').toMatchSnapshot();
    });
  })

    describe('Data-Driven Test', () => {
      beforeEach(() => {
        pageObjects.actions.goToAppUrl();
      });
      it('cannot login user without both username and password', () => {
        pageObjects.actions.clickLoginButton();
        pageObjects.expects.matchImageSnapshot('.alert-warning', 'cannot login user without both username and password')
        pageObjects.elements.alertWarning().contains("Both Username and Password must be present");
      });
      it('cannot login user without password', () => {
        pageObjects.actions.populateUsernameField('test');
        pageObjects.actions.clickLoginButton();
        pageObjects.expects.matchImageSnapshot('.alert-warning', 'cannot login user without password')
        pageObjects.elements.alertWarning().contains('Password must be present');
      });

      it('cannot login user without username', () => {
        pageObjects.actions.populatePasswordField('testPassword');
        pageObjects.actions.clickLoginButton();
        pageObjects.expects.matchImageSnapshot('.alert-warning', 'cannot login user without username')
        pageObjects.elements.alertWarning().contains('Username must be present');
      });

      it('can successfully log in user', () => {
        pageObjects.actions.populateUsernameField('test');
        pageObjects.actions.populatePasswordField('testPassword');
        pageObjects.actions.clickLoginButton();
        pageObjects.expects.dashboardIsPresent();
      });

      it('can successfully log in user with different login details', () => {
        pageObjects.actions.populateUsernameField('TestUser');
        pageObjects.actions.populatePasswordField('AnotherTestPassword');
        pageObjects.actions.clickLoginButton();
        pageObjects.expects.dashboardIsPresent();
      });
    });

    describe('Table Sort Test', () => {
      beforeEach(() => {
        pageObjects.actions.goToAppUrl();
        cy.login('test', 'user123');
      });
      it('can confirm amount details are correct and consistent over time', () => {
         pageObjects.actions.clickAmountHeader()
         cy.get('#transactionsTable').toMatchSnapshot();
      });
      it('can verify amounts are sorted in ascending order', () => {
        pageObjects.expects.starbucksCoffeeIsPresent({ positionId: 1 });
        pageObjects.expects.stripePaymentProcessingIsPresent({ positionId: 2 });
        pageObjects.expects.mailChimpServicesIsPresent({ positionId: 3 });
        pageObjects.expects.shopifyProductIsPresent({ positionId: 4 });
        pageObjects.expects.ebayMarketplaceIsPresent({ positionId: 5 });
        pageObjects.expects.templatesIncIsPresent({ positionId: 6 });

        pageObjects.actions.clickAmountHeader()

        pageObjects.expects.mailChimpServicesIsPresent({ positionId: 1 });
        pageObjects.expects.ebayMarketplaceIsPresent({ positionId: 2 });
        pageObjects.expects.shopifyProductIsPresent({ positionId: 3 });
        pageObjects.expects.templatesIncIsPresent({ positionId: 4 });
        pageObjects.expects.stripePaymentProcessingIsPresent({ positionId: 5 });
        pageObjects.expects.starbucksCoffeeIsPresent({ positionId: 6 });
     });

    
    });

    describe('Canvas Chart Test', () => {
      beforeEach(() => {
        pageObjects.actions.goToAppUrl();
        cy.login('test', 'user123');
    });

    it('can show expenses and forecast comparison chart consistency in different app versions',  () => {
       pageObjects.actions.clickCompareExpensesButton();
       pageObjects.expects.canvasChartIsPresent();
       pageObjects.expects.matchImageSnapshot('#canvas', 'expenses and forecast chart')
    });

    it('can show expenses and forecast comparison chart with 2019 comparison',  () => {
      pageObjects.actions.clickCompareExpensesButton();
      pageObjects.expects.canvasChartIsPresent();
      pageObjects.actions.clickDatasetButton();
      pageObjects.expects.matchImageSnapshot('#canvas', 'expenses and forecast chart with 2019 comparison');
   });

    });

    describe.only('Dynamic Content Test', () => {
      beforeEach(() => {
        pageObjects.actions.goToAppUrlWithAds();
        cy.login('test', 'user123');
    });
    
    it('can verify the ad section is visible', () => {
      pageObjects.expects.dashboardIsPresent();
      pageObjects.expects.firstAdSectionIsVisible();
      pageObjects.expects.secondAdSectionIsVisible();
      cy.get('.element-balances').toMatchSnapshot();
    });

    it('can verify that ad can change and be dynamic', () => {
      // currently unable to expect an unknown element id and snapshot it in cypress to ensure the test passes
      // We cannot mock an element that does not exist in our application yet
    })
  });
  

  afterEach(() => {
     cy.logout('test');
  });
});
});
