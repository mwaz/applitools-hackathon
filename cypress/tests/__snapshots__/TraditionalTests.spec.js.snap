exports[`ACME app > ACME app User login > Table Sort Test > can verify amounts are sorted in ascending order #0`] = `
<table id="transactionsTable" class="table table-padded">
  <thead>
    <tr>
      <th
        id="status"
        onclick="sortTable(0)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Status
      </th>
      <th
        id="date"
        onclick="sortTable(1)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Date
      </th>
      <th
        id="description"
        onclick="sortTable(2)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Description
      </th>
      <th
        id="category"
        onclick="sortTable(3)"
        class="text-center"
        onmouseover=""
        style="cursor: pointer;"
      >
        Category
      </th>
      <th
        id="amount"
        onclick="sortTable(4)"
        class="text-right"
        onmouseover=""
        style="cursor: pointer;"
      >
        Amount
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Yesterday</span><span class="smaller lighter">7:45am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company3.png" style="height: 25px;" /><span
          >MailChimp Services</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-warning" href="">Software</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-danger">- 320.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller green"></span><span>Complete</span>
      </td>
      <td><span>Jan 7th</span><span class="smaller lighter">9:51am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company4.png" style="height: 25px;" /><span
          >Ebay Marketplace</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-danger" href="">Ecommerce</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-danger">- 244.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Jan 23rd</span><span class="smaller lighter">2:7pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company6.png" style="height: 25px;" /><span
          >Shopify product</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-primary" href="">Shopping</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 17.99 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Jan 9th</span><span class="smaller lighter">7:45pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company7.png" style="height: 25px;" /><span>
          Templates Inc</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-primary" href="">Business</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 340.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller red"></span><span>Declined</span>
      </td>
      <td><span>Jan 19th</span><span class="smaller lighter">3:22pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company2.png" style="height: 25px;" /><span
          >Stripe Payment Processing</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-danger" href="">Finance</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 952.23 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller green"></span><span>Complete</span>
      </td>
      <td><span>Today</span><span class="smaller lighter">1:52am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company1.png" style="height: 25px;" /><span
          >Starbucks coffee</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-success" href="">Restaurant / Cafe</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 1,250.00 USD</span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`ACME app > ACME app User login > Table Sort Test > can verify amounts are sorted in ascending order using snapshots #0`] = `
<table id="transactionsTable" class="table table-padded">
  <thead>
    <tr>
      <th
        id="status"
        onclick="sortTable(0)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Status
      </th>
      <th
        id="date"
        onclick="sortTable(1)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Date
      </th>
      <th
        id="description"
        onclick="sortTable(2)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Description
      </th>
      <th
        id="category"
        onclick="sortTable(3)"
        class="text-center"
        onmouseover=""
        style="cursor: pointer;"
      >
        Category
      </th>
      <th
        id="amount"
        onclick="sortTable(4)"
        class="text-right"
        onmouseover=""
        style="cursor: pointer;"
      >
        Amount
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Yesterday</span><span class="smaller lighter">7:45am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company3.png" style="height: 25px;" /><span
          >MailChimp Services</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-warning" href="">Software</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-danger">- 320.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller green"></span><span>Complete</span>
      </td>
      <td><span>Jan 7th</span><span class="smaller lighter">9:51am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company4.png" style="height: 25px;" /><span
          >Ebay Marketplace</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-danger" href="">Ecommerce</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-danger">- 244.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Jan 23rd</span><span class="smaller lighter">2:7pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company6.png" style="height: 25px;" /><span
          >Shopify product</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-primary" href="">Shopping</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 17.99 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Jan 9th</span><span class="smaller lighter">7:45pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company7.png" style="height: 25px;" /><span>
          Templates Inc</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-primary" href="">Business</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 340.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller red"></span><span>Declined</span>
      </td>
      <td><span>Jan 19th</span><span class="smaller lighter">3:22pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company2.png" style="height: 25px;" /><span
          >Stripe Payment Processing</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-danger" href="">Finance</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 952.23 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller green"></span><span>Complete</span>
      </td>
      <td><span>Today</span><span class="smaller lighter">1:52am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company1.png" style="height: 25px;" /><span
          >Starbucks coffee</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-success" href="">Restaurant / Cafe</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 1,250.00 USD</span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`ACME app > ACME app User login > Table Sort Test > can confirm #0`] = `
<table id="transactionsTable" class="table table-padded">
  <thead>
    <tr>
      <th
        id="status"
        onclick="sortTable(0)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Status
      </th>
      <th
        id="date"
        onclick="sortTable(1)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Date
      </th>
      <th
        id="description"
        onclick="sortTable(2)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Description
      </th>
      <th
        id="category"
        onclick="sortTable(3)"
        class="text-center"
        onmouseover=""
        style="cursor: pointer;"
      >
        Category
      </th>
      <th
        id="amount"
        onclick="sortTable(4)"
        class="text-right"
        onmouseover=""
        style="cursor: pointer;"
      >
        Amount
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Yesterday</span><span class="smaller lighter">7:45am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company3.png" style="height: 25px;" /><span
          >MailChimp Services</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-warning" href="">Software</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-danger">- 320.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller green"></span><span>Complete</span>
      </td>
      <td><span>Jan 7th</span><span class="smaller lighter">9:51am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company4.png" style="height: 25px;" /><span
          >Ebay Marketplace</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-danger" href="">Ecommerce</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-danger">- 244.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Jan 23rd</span><span class="smaller lighter">2:7pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company6.png" style="height: 25px;" /><span
          >Shopify product</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-primary" href="">Shopping</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 17.99 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Jan 9th</span><span class="smaller lighter">7:45pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company7.png" style="height: 25px;" /><span>
          Templates Inc</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-primary" href="">Business</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 340.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller red"></span><span>Declined</span>
      </td>
      <td><span>Jan 19th</span><span class="smaller lighter">3:22pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company2.png" style="height: 25px;" /><span
          >Stripe Payment Processing</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-danger" href="">Finance</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 952.23 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller green"></span><span>Complete</span>
      </td>
      <td><span>Today</span><span class="smaller lighter">1:52am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company1.png" style="height: 25px;" /><span
          >Starbucks coffee</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-success" href="">Restaurant / Cafe</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 1,250.00 USD</span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`ACME app > ACME app User login > Table Sort Test > can confirm amount details are correct and consistent over time #0`] = `
<table id="transactionsTable" class="table table-padded">
  <thead>
    <tr>
      <th
        id="status"
        onclick="sortTable(0)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Status
      </th>
      <th
        id="date"
        onclick="sortTable(1)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Date
      </th>
      <th
        id="description"
        onclick="sortTable(2)"
        onmouseover=""
        style="cursor: pointer;"
      >
        Description
      </th>
      <th
        id="category"
        onclick="sortTable(3)"
        class="text-center"
        onmouseover=""
        style="cursor: pointer;"
      >
        Category
      </th>
      <th
        id="amount"
        onclick="sortTable(4)"
        class="text-right"
        onmouseover=""
        style="cursor: pointer;"
      >
        Amount
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Yesterday</span><span class="smaller lighter">7:45am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company3.png" style="height: 25px;" /><span
          >MailChimp Services</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-warning" href="">Software</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-danger">- 320.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller green"></span><span>Complete</span>
      </td>
      <td><span>Jan 7th</span><span class="smaller lighter">9:51am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company4.png" style="height: 25px;" /><span
          >Ebay Marketplace</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-danger" href="">Ecommerce</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-danger">- 244.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Jan 23rd</span><span class="smaller lighter">2:7pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company6.png" style="height: 25px;" /><span
          >Shopify product</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-primary" href="">Shopping</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 17.99 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller yellow"></span><span>Pending</span>
      </td>
      <td><span>Jan 9th</span><span class="smaller lighter">7:45pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company7.png" style="height: 25px;" /><span>
          Templates Inc</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-primary" href="">Business</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 340.00 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller red"></span><span>Declined</span>
      </td>
      <td><span>Jan 19th</span><span class="smaller lighter">3:22pm</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company2.png" style="height: 25px;" /><span
          >Stripe Payment Processing</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-danger" href="">Finance</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 952.23 USD</span>
      </td>
    </tr>
    <tr>
      <td class="nowrap">
        <span class="status-pill smaller green"></span><span>Complete</span>
      </td>
      <td><span>Today</span><span class="smaller lighter">1:52am</span></td>
      <td class="cell-with-media">
        <img alt="" src="img/company1.png" style="height: 25px;" /><span
          >Starbucks coffee</span
        >
      </td>
      <td class="text-center">
        <a class="badge badge-success" href="">Restaurant / Cafe</a>
      </td>
      <td class="text-right bolder nowrap">
        <span class="text-success">+ 1,250.00 USD</span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`ACME app > ACME app User login > Login Page UI Elements Test: > can get correct a snapshot of the login page #0`] = `
<div class="logo-w">
  <a href="index.html"><img alt="" src="img/logo-big.png"/></a>
</div>
`;

exports[`ACME app > ACME app User login > Login Page UI Elements Test: > can get correct a snapshot of the login page #1`] = `
[ {}, {} ]
`;

exports[`ACME app > ACME app User login > Login Page UI Elements Test: > can get correct a snapshot of the login page #2`] = `
<h4 class="auth-header">
  Login Form
</h4>
`;

exports[`ACME app > ACME app User login > Login Page UI Elements Test: > can get correct a snapshot of the logo section #0`] = `
<div class="logo-w">
  <a href="index.html"><img alt="" src="img/logo-big.png"/></a>
</div>
`;

exports[`ACME app > ACME app User login > Login Page UI Elements Test: > can get correct a snapshot of the form group #0`] = `
[ {}, {} ]
`;

exports[`ACME app > ACME app User login > Login Page UI Elements Test: > can get correct a snapshot of the auth header #0`] = `
<h4 class="auth-header">
  Login Form
</h4>
`;

exports[`ACME app > ACME app User login > Login Page UI Elements Test: > can get correct a snapshot of the button group #0`] = `
<div class="buttons-w">
  <button id="log-in" class="btn btn-primary">Log In</button>
  <div class="form-check-inline">
    <label class="form-check-label"
      ><input class="form-check-input" type="checkbox" />Remember Me</label
    >
  </div>
  <div style="text-align:center">
    <a href="#" style="display: inline-block; margin-bottom:4px;"
      ><img alt="" src="img/social-icons/twitter.png" style="width: 28px;"/></a
    ><a
      href="#"
      style="display: inline-block; margin-left: 30px;margin-right:30px;margin-top:20px;"
      ><img alt="" src="img/social-icons/facebook.png" style="width: 28px;"/></a
    ><a
      href="#"
      style="display: inline-block; margin-left: 30px;margin-right:30px;margin-top:20px;"
      ><img alt="" src="img/social-icons/linkedin.png" style="width: 28px;"
    /></a>
  </div>
</div>
`;

exports[`ACME app > ACME app User login > Data-Driven Test > cannot login user without password #0`] = `
<div
  id="random_id_3"
  class="alert alert-warning"
  role="alert"
  style="display: block;"
>
  Password must be present
</div>
`;

exports[`ACME app > ACME app User login > Data-Driven Test > cannot login user without both username and password #0`] = `
<div
  id="random_id_1"
  class="alert alert-warning"
  role="alert"
  style="display: block;"
>
  Both Username and Password must be present
</div>
`;

exports[`ACME app > ACME app User login > Data-Driven Test > cannot login user without username #0`] = `
<div
  id="random_id_7"
  class="alert alert-warning"
  role="alert"
  style="display: block;"
>
  Username must be present
</div>
`;

exports[`ACME app > ACME app User login > Dynamic Content Test > can verify the ad section is visible #0`] = `
<div class="element-balances">
  <div class="balance hidden-mobile" id="totalBalance" style="display: none;">
    <div class="balance-title">
      Total Balance
    </div>
    <div class="balance-value">
      <span>$350</span
      ><span class="trending trending-down-basic"
        ><span>%7</span><i class="os-icon os-icon-arrow-2-down"></i
      ></span>
    </div>
    <div class="balance-link">
      <a class="btn btn-link btn-underlined" href="#"
        ><span>View Statement</span><i class="os-icon os-icon-arrow-right4"></i
      ></a>
    </div>
  </div>
  <div class="balance hidden-mobile" id="flashSale" style="display: block;">
    <img alt="" src="img/flashSale.gif" style="width:129px;height:100px" />
  </div>
  <div class="balance" id="creditAvailable" style="display: none;">
    <div class="balance-title">
      Credit Available
    </div>
    <div class="balance-value">
      $17,800
    </div>
    <div class="balance-link">
      <a class="btn btn-link btn-underlined" href="#"
        ><span>Request Increase</span
        ><i class="os-icon os-icon-arrow-right4"></i
      ></a>
    </div>
  </div>
  <div class="balance hidden-mobile" id="flashSale2" style="display: block;">
    <img alt="" src="img/flashSale2.gif" style="width:140px;height:100px;" />
  </div>
  <div class="balance">
    <div class="balance-title">
      Due Today
    </div>
    <div class="balance-value danger">
      $180
    </div>
    <div class="balance-link">
      <a class="btn btn-link btn-underlined btn-gold" href="#"
        ><span>Pay Now</span><i class="os-icon os-icon-arrow-right4"></i
      ></a>
    </div>
  </div>
</div>
`;
